policy/pipeline/pipeline_version.rego

```diff
@@ -0,0 +1,68 @@
+package policy.pipeline.pipeline_version
+
+import data.lib
+
+VERSION_REGEX := `^[0-9]+(\.[0-0]+){0,2}`
+
+# METADATA
+# title: Pipeline defines a version
+# description: |-
+#   The Pipeline has an explicitly defined version value. The version is
+#   expected to be defined by the "app.kubernetes.io/version" resource label.
+#
+# custom:
+#   short_name: pipeline_version_exists
+#   failure_msg: Pipeline %q does not define a version
+#   effective_on: 2023-01-01T00:00:00Z
+#
+deny[result] {
+	pipeline_name := input.metadata.name
+
+	not input.metadata.labels["app.kubernetes.io/version"]
+
+	result := lib.result_helper(rego.metadata.chain(), [pipeline_name])
+}
+
+# METADATA
+# title: Pipeline version must be in a specific format
+# description: |-
+#   The Pipeline version is expected to be in one of the following formats:
+#   1
+#   1.2
+#   1.2.3
+#
+#   Each part of the version must have 1 or more digits. All other characters are
+#   not allowed.
+#
+# custom:
+#   short_name: pipeline_version_format
+#   failure_msg: Pipeline %q defines an invalid version format, %s. Use "." separated digits
+#   effective_on: 2023-01-01T00:00:00Z
+#
+deny[result] {
+	pipeline_name := input.metadata.name
+	version := input.metadata.labels["app.kubernetes.io/version"]
+	not regex.match(VERSION_REGEX, version)
```

Comment:
We could use the [semver](https://www.openpolicyagent.org/docs/latest/policy-reference/#semver) functions from OPA
```suggestion
	not semver.is_valid(version)
```